File: 2.Dagger/AndroidInjector/AppActivityHierarchyModified/src/main/java/com/app/modifiedappactivityhierarchy/activity/MainActivity.java
package com.app.modifiedappactivityhierarchy.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import com.app.modifiedappactivityhierarchy.R;

import javax.inject.Inject;

import dagger.android.AndroidInjection;

public class MainActivity extends AppCompatActivity {
    @Inject
    String str;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.i(TAG, str);
    }

    private static final String TAG = "MainActivity";
}

File: 2.Dagger/Hilt/app/src/main/java/com/app/hilt/MainActivity.kt
package com.app.hilt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.activity.viewModels
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var a:A

    val mainViewModel: MainViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.i(TAG, mainViewModel.dep.toString())
    }

    companion object {
        private const val TAG = "MainActivity"
    }
}